frameworks/autosklearn/setup.sh

```diff
@@ -19,13 +19,28 @@ PIP install --no-cache-dir -r $HERE/requirements.txt
 if [[ "$VERSION" == "stable" ]]; then
     PIP install --no-cache-dir -U ${PKG}
 elif [[ "$VERSION" =~ ^[0-9] ]]; then
+    # Provided a version number
     PIP install --no-cache-dir -U ${PKG}==${VERSION}
 else
-#    PIP install --no-cache-dir -e git+${REPO}@${VERSION}#egg=${PKG}
     TARGET_DIR="${HERE}/lib/${PKG}"
     rm -Rf ${TARGET_DIR}
-    git clone --depth 1 --single-branch --branch ${VERSION} --recurse-submodules ${REPO} ${TARGET_DIR}
+
+    if [[ "$VERSION" =~ ^# ]]; then
+      COMMIT="${VERSION:1}"
+    else
+      # find the latest commit to the VERSION branch
+      COMMIT=$(git ls-remote "${REPO}" | grep "refs/heads/${VERSION}" | cut -f 1)
```

Comment:
why to we need the commit when we specify a branch?
Then it forces us to clone the entire repository when we used to need to clone only the specific branch without history... it's a bit silly to clone a full repo when you're not planning to work on it.

Reply:
ok, I guess this optimization was temporarily removed.